internal/rbd/encryption.go

```diff
@@ -463,6 +465,34 @@ func (rv *rbdVolume) RotateEncryptionKey(ctx context.Context) error {
 		return errors.New("key rotation not supported for unencrypted device")
 	}
 
+	// Call open Ioctx to create a new ioctx object
+	// if the obj already exists, no error is returned
+	err = rv.openIoctx()
+	if err != nil {
+		log.ErrorLogMsg("failed to open ioctx, err: %v", err)
+
+		return fmt.Errorf("failed to open ioctx, err: %w", err)
+	}
+
+	// Lock params
+	lockName := rv.VolID + "-mutexlock"
+	lockDesc := "Key rotation mutex lock for " + rv.VolID
+	lockDuration := 3 * time.Minute
+	lockCookie := rv.VolID + "-enc-key-rotate"
+	var flags byte = 0
+
+	// Acquire the exclusive lock based on vol id
+	lck := lock.NewLock(rv.ioctx, rv.VolID, lockName, lockCookie, lockDesc, lockDuration, &flags)
+	err = lck.LockExclusive(ctx)
+	//nolint:errcheck // No need to log here again
+	defer lck.Unlock(ctx)
```

Comment:
we need to move Unlock after err check or else we might release the lock held by someone else.

Reply:
> or else we might release the lock held by someone else.

If a lock with name/cookie pair does not exist, it will return `ENOENT`, which in turn will be logged. The unlock call is not a direct call to ioctx's unlock but rather a wrapper around it.

Reply:
Ack.